Source/Dafny/Printer.cs

```diff
@@ -2362,6 +2360,10 @@ void PrintExpr(Expression expr, int contextBindingStrength, bool fragileContext,
         // TODO should we also print the desugared version?
         // If so, should we insert newlines?
         var e = (LetOrFailExpr)expr;
+        if(e.ResolvedExpression != null){
```

Comment:
Here and everywhere, put a space after the `if` and a space before the `{` at the end of the line.